### ISSUE TITLE
InfluxDb certificate error when upgrade from 8.0.2 to 8.1.0

### ISSUE THREAD

**Issue by Oscar-Young** [grafana/grafana#37683]
**What happened:**
Upgrade from 8.0.2 to 8.1.0, And Alert chart using influxdb  to report the `certificate signed by unknown authority`
When I rolled back to version 8.0.2, everything was fine.

**What you expected to happen:**
I want to understand what happened.

**How to reproduce it (as minimally and precisely as possible):**

- Alarm chart using influxdb.
- Influxdb : use certificates or expired certificates ( or other ? )

**Environment:**

docker-compose.yaml:

```YAML
version: "3"
services:
  grafana:
    container_name: grafana
    image: grafana/grafana:8.0.2
    user: "0"
    ports:
      - "5001:5001"
    volumes:
      - ./grafana:/var/lib/grafana
      - ./etc/grafana:/etc/grafana
```

**Comment by N3v3R3nD** [comment]
Could it be related to this #37177 ?

**Comment by dsotirakis** [comment]
> Could it be related to this #37177 ?
I think yes, looping in @grafana/observability-squad

**Comment by gabor** [comment]
hi @Oscar-Young , could you provide some more info please? 
- is this happening only for alerts or also normal queries, like in dashboards?
- what version of influxdb are you using?
- are you connecting to influxdb in "Flux" mode or in "InfluxQL" mode?
- if you go to the datasource configuration page and press the [Save & test] button at the bottom, does that work or that fails too?
- how is your influxdb datasource configured? (please remove any private/secret info)
  - are you using a `https://` URL?
  - basically, tell us, in the influxdb-datasource-config, which fields are used (which "checkboxes are set", which fields contain info)
 
thanks!

**Comment by krombel** [comment]
We have the same issue in our infra:
- This is only an issue for alerts
- we are running on influx 1.8

![Screenshot 2021-08-10 at 21 23 16](https://user-images.githubusercontent.com/11167142/128922288-0a935429-0b70-459b-af20-d7851813770a.png)

![Screenshot 2021-08-10 at 21 20 01](https://user-images.githubusercontent.com/11167142/128921986-68d000fb-05bb-4477-87bd-dcc25a37fd8c.png)

I hope this help fix this issue

**Comment by DasSkelett** [comment]
Don't quote me on this, but after looking around the code and commits a bit, I suspect this is "caused" by https://github.com/grafana/grafana/commit/cec12676e7727412f97d32330ca9f6b54e83ae0b (#36189) and the switch to the "backend SDK".

tl;dr: The "Skip TLS Verify" option is now ignored unless "With CA Cert" or "TLS Client Auth" are set.

However setting "With CA Cert" without pasting an actual cert as a workaround doesn't work and triggers the 401 Unauthorized bug mentioned in #37686.

---

Before, an HTTP client was received using `ds.GetHTTPClient()` from `datasource_cache.go`:
https://github.com/grafana/grafana/blob/27b7c35ccc1bd4a61473255ea1e11836d39b114a/pkg/tsdb/influxdb/influxdb.go#L82

which called `ds.GetHTTPTransport()`:
https://github.com/grafana/grafana/blob/27b7c35ccc1bd4a61473255ea1e11836d39b114a/pkg/models/datasource_cache.go#L50

which called `ds.HTTPClientOptions()`
https://github.com/grafana/grafana/blob/27b7c35ccc1bd4a61473255ea1e11836d39b114a/pkg/models/datasource_cache.go#L69

which got its TLS configuration from `ds.TLSOptions()`:
https://github.com/grafana/grafana/blob/27b7c35ccc1bd4a61473255ea1e11836d39b114a/pkg/models/datasource_cache.go#L86

which unconditionally set `InsecureSkipVerify` if it is set in the data source settings:
https://github.com/grafana/grafana/blob/27b7c35ccc1bd4a61473255ea1e11836d39b114a/pkg/models/datasource_cache.go#L153-L158

---

Now, it uses `dsInfo.HTTPClient.Do()`:
https://github.com/grafana/grafana/blob/cec12676e7727412f97d32330ca9f6b54e83ae0b/pkg/tsdb/influxdb/influxdb.go#L145

This HTTPClient is created in `newInstanceSettings()`:
https://github.com/grafana/grafana/blob/cec12676e7727412f97d32330ca9f6b54e83ae0b/pkg/tsdb/influxdb/influxdb.go#L70-L75
https://github.com/grafana/grafana/blob/cec12676e7727412f97d32330ca9f6b54e83ae0b/pkg/tsdb/influxdb/influxdb.go#L93-L94

which gets its options from `HTTPClientOptions()` in github.com/grafana/grafana-plugin-sdk-go `backend/common.go`
https://github.com/grafana/grafana-plugin-sdk-go/blob/c0722b0e1babfd61eb18828b9364dfb7d05d0e12/backend/common.go#L94-L95

and  `parseHTTPSettings()` in `http_settings.go`, **which only respects the tlsSkipVerify option if tlsClientAuth or tlsAuthWithCACerrt are set**:
https://github.com/grafana/grafana-plugin-sdk-go/blob/c0722b0e1babfd61eb18828b9364dfb7d05d0e12/backend/http_settings.go#L221-L224

I am not sure if this is intended or not, but it definitely is breaking behavior. The new client and the old client read the configuration differently, and thus behave differently.

**Comment by Oscar-Young** [comment]
Hi @gabor ,
Yes i'm using HTTPS and My setting are the same as @kakkoyun , 

And my alarm chart looks the same as his #37687

**Comment by gabor** [comment]
@DasSkelett you are very probably right, many datasources switched to this new approach, and there's a very good chance this is causing it. my only problem is, i cannot reproduce it easily, see my comment in https://github.com/grafana/grafana/issues/37177#issuecomment-895820912

but we will look into this more.

**Comment by gabor** [comment]
@DasSkelett ignore my previous comment, my mistake. yes, i can reproduce this (the error does not happen when you press the [save&test] button in the datasource, or if you make a query (because that is frontend-queried), it only happens in an alert.

**Comment by gabor** [comment]
so, for reference, here is how to reproduce it with influxdb:

- setup an influxdb2 database using the docs from https://docs.influxdata.com/influxdb/v2.0/security/enable-tls/#configure-influxdb-to-use-tls , creating a self-signed certificate using the `openssl` command mentioned there
- add an influxdb datasource in influxql (!!this is important!!) mode using `https://`
- enable "skip tls verify"
- when you press the [save&test] button, it will work
- (for the following steps we assume new-alert-mode. it happens in old-alert-mode too)
- go create a new alert (grafana managed alert)
- choose the just-created data-source
- in the query editor, open the "select measurement" dropdown, and select something
- now you get the error-message

**Comment by gabor** [comment]
fixed by https://github.com/grafana/grafana/pull/37797 , closing issue.
